xde/src/route.rs

```diff
@@ -0,0 +1,587 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+use crate::ip;
+use crate::sys;
+use crate::xde::xde_underlay_port;
+use crate::xde::DropRef;
+use crate::xde::XdeDev;
+use alloc::collections::btree_map::Entry;
+use alloc::collections::BTreeMap;
+use alloc::sync::Arc;
+use core::ptr;
+use core::time::Duration;
+use illumos_sys_hdrs::*;
+use opte::ddi::sync::KRwLock;
+use opte::ddi::sync::KRwLockType;
+use opte::ddi::time::Moment;
+use opte::engine::ether::EtherAddr;
+use opte::engine::ip6::Ipv6Addr;
+
+// XXX: completely arbitrary timeouts.
+/// The duration a cached route remains valid for before it must be
+/// refreshed.
+///
+/// Expired routes will not be removed from the cache, and will leave
+/// an entry to enable a quick in-place refresh in the `BTreeMap`.
+const EXPIRE_ROUTE_LIFETIME: Duration = Duration::from_millis(100);
```

Comment:
For the moment, this is well within the DDM control plane neighbor expiry lifetime. We'll revisit when the DDM data plane lands.